src/controllers/expenses.js

```diff
@@ -0,0 +1,144 @@
+'use strict';
+
+const {
+  getMany: getManyExpenses,
+  getOne: getOneExpense,
+  create: createExpense,
+  remove: removeExpense,
+  update: updateExpense,
+  normalize,
+} = require('../services/expenses');
+
+const { getOne: getOneUser } = require('../services/users');
+
+const getMany = async(req, res) => {
+  const {
+    userId,
+    category,
+    from,
+    to,
+  } = req.query;
+
+  const dateIsInvalid
+  = (userId && isNaN(+userId))
+    || (from && isNaN(Date.parse(from)))
+    || (to && isNaN(Date.parse(to)))
+    || (category && typeof category !== 'string');
+
+  if (dateIsInvalid) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const filteredExpenses = await getManyExpenses(userId, category, from, to);
+
+  res.send(filteredExpenses.map(normalize));
+};
+
+const getOne = async(req, res) => {
+  const { expenseId } = req.params;
+
+  if (isNaN(+expenseId)) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const foundExpense = await getOneExpense(expenseId);
+
+  foundExpense ? res.send(normalize(foundExpense)) : res.sendStatus(404);
+};
+
```

Comment:
you should send status code 200 on success

Reply:
But we have this status by default 🤔